WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.kt

```diff
@@ -0,0 +1,130 @@
+        initSite(savedInstanceState)
+        val builder =
+            MaterialAlertDialogBuilder(ContextThemeWrapper(activity, R.style.PostSettingsTheme))
+        binding = AddCategoryBinding.inflate(layoutInflater, null, false)
+        loadCategories()
+        builder.setView(binding!!.root)
+            .setPositiveButton(android.R.string.ok, null)
+            .setNegativeButton(android.R.string.cancel, null)
+
+        return builder.create()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val dialog = requireDialog() as AlertDialog
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener {
+            if (addCategory()) {
+                dismiss()
+            }
+        }
+    }
+
+    @Suppress("Deprecation")
+    private fun initSite(savedInstanceState: Bundle?) {
+        site = if (savedInstanceState == null) {
+            if (arguments != null) {
+                requireArguments().getSerializable(WordPress.SITE) as SiteModel?
+            } else {
+                requireActivity().intent.getSerializableExtra(WordPress.SITE) as SiteModel?
+            }
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel?
+        }
+
+        if (site == null) {
+            ToastUtils.showToast(
+                requireActivity(),
+                R.string.blog_not_found,
+                ToastUtils.Duration.SHORT
+            )
+            parentFragmentManager.popBackStack()
+        }
+    }
+
+    private fun addCategory(): Boolean {
+        val categoryName = binding?.categoryName?.text.toString()
+        val selectedCategory = binding?.parentCategory?.selectedItem as CategoryNode
+        val parentId = selectedCategory.categoryId
+
+        if (categoryName.replace(" ".toRegex(), "") == "") {
+            binding!!.categoryName.error = getText(R.string.cat_name_required)
+            return false
+        }
+
+        val newCategory = TermModel(
+            TaxonomyStore.DEFAULT_TAXONOMY_CATEGORY,
+            categoryName,
+            parentId
+        )
+        (requireActivity() as SelectCategoriesActivity).categoryAdded(newCategory)
+        return true
+    }
+
+    private fun loadCategories() {
+        val rootCategory = CategoryNode.createCategoryTreeFromList(
+            mTaxonomyStore!!.getCategoriesForSite(
+                site!!
+            )
+        )
+        val categoryLevels = CategoryNode.getSortedListOfCategoriesFromRoot(rootCategory)
+        categoryLevels.add(0, CategoryNode(0, 0, getString(R.string.top_level_category_name)))
+        if (categoryLevels.size > 0) {
+            val categoryAdapter =
+                ParentCategorySpinnerAdapter(
+                    activity,
+                    R.layout.categories_row_parent,
+                    categoryLevels
+                )
+            binding!!.parentCategory.adapter = categoryAdapter
```

Comment:
Please update here with `binding?.parentCategory?.adapter = categoryAdapter` to avoid `!!`.

Reply:
Fixed in the latest commit 👍